src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -672,23 +672,13 @@ public override void DidCompleteWithError (NSUrlSession session, NSUrlSessionTas
 
 			void SetResponse (InflightData inflight)
 			{
-				lock (inflight.Lock) {
-					if (inflight.ResponseSent)
-						return;
-
-					if (inflight.CancellationTokenSource.Token.IsCancellationRequested)
-						return;
-
-					if (inflight.CompletionSource.Task.IsCompleted)
-						return;
-
-					var httpResponse = inflight.Response;
+				if (inflight.CancellationTokenSource.Token.IsCancellationRequested)
+					return;
 
-					inflight.ResponseSent = true;
+				if (inflight.CompletionSource.Task.IsCompleted)
+					return;
 
-					// EVIL HACK: having TrySetResult inline was blocking the request from completing
-					Task.Run (() => inflight.CompletionSource.TrySetResult (httpResponse));
```

Comment:
I don't understand why this is leaking. The closure will be collected by the GC when the Task.Run method completes.

Reply:
The closure should be GC, but it is not. Look at https://github.com/xamarin/xamarin-macios/issues/8350#issuecomment-626106244 profiling results. The leak is a memory 
stream from the httpcontent of the response. Removing the use of a lambda, fixes the issue.

The leak only happens when it is used in a sync method, so in async the lambda is correctly GC, in sync is not. I could not pin point the exact reason on why the GC is not doing the job correctly. 

The code was wrong nevertheless.